addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/FreeboxAirPlayAudioSink.java

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * This makes a AirPlay device to serve as an {@link AudioSink}-
+ *
+ * @author Laurent Garnier - Initial contribution for AudioSink and notifications
+ */
+public class FreeboxAirPlayAudioSink implements AudioSink {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeboxAirPlayAudioSink.class);
+
+    private static final AudioFormat MP3_96 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 96000, null);
+    private static final AudioFormat MP3_112 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 112000, null);
+    private static final AudioFormat MP3_128 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 128000, null);
+    private static final AudioFormat MP3_160 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 160000, null);
+    private static final AudioFormat MP3_192 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 192000, null);
+    private static final AudioFormat MP3_224 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 224000, null);
+    private static final AudioFormat MP3_256 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 256000, null);
+    private static final AudioFormat MP3_320 = new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, null, 320000, null);
+
+    private Set<AudioFormat> supportedFormats;
+    private AudioHTTPServer audioHTTPServer;
+    private FreeboxThingHandler handler;
+    private String callbackUrl;
+
+    public FreeboxAirPlayAudioSink(FreeboxThingHandler handler, AudioHTTPServer audioHTTPServer, String callbackUrl) {
+        this.handler = handler;
+        this.audioHTTPServer = audioHTTPServer;
+        this.callbackUrl = callbackUrl;
+        Boolean acceptLowBitrate = (Boolean) handler.getThing().getConfiguration()
+                .get(FreeboxAirPlayDeviceConfiguration.ACCEPT_ALL_MP3);
+        this.supportedFormats = new HashSet<>();
+        this.supportedFormats.add(WAV);
+        if (acceptLowBitrate) {
+            this.supportedFormats.add(MP3);
+        } else {
+            // Only accept MP3 bitrates >= 96 kbps
+            this.supportedFormats.add(MP3_96);
+            this.supportedFormats.add(MP3_112);
+            this.supportedFormats.add(MP3_128);
+            this.supportedFormats.add(MP3_160);
+            this.supportedFormats.add(MP3_192);
+            this.supportedFormats.add(MP3_224);
+            this.supportedFormats.add(MP3_256);
+            this.supportedFormats.add(MP3_320);
+        }
+        this.supportedFormats.add(OGG);
+    }
+
+    @Override
+    public String getId() {
+        return handler.getThing().getUID().toString();
+    }
+
+    @Override
+    public String getLabel(Locale locale) {
+        return handler.getThing().getLabel();
+    }
+
+    @Override
+    public void process(AudioStream audioStream) throws UnsupportedAudioFormatException {
+        if (!ThingHandlerHelper.isHandlerInitialized(handler)
+                || (handler.getThing().getStatus() != ThingStatus.ONLINE)) {
```

Comment:
Want to check at least that the bridge is not OFFLINE and that the thing is not in configuration error.